### DISCUSSION TITLE
API Key in Docker Container

### DISCUSSION THREAD

**Discussion by lixsldotnet** [simulationcraft/simc#7705] [Q&A]
Im building a small scale applications for myself at the moment in NodeJS but i cant figure out where to put the .simc_apikey file in the docker container.

I allways get the same Timeout Error:

Error: Option 'armory' with value 'eu,Blackrock,Stabix': BCP API: Unable to download JSON from 'https://eu.api.blizzard.com/profile/wow/character/blackrock/stabix?namespace=profile-eu&locale=en_GB'.: Unable to authorize: Unable to fetch bearer token from https://eu.battle.net/oauth/token, response=0, error=Operation timed out after 15001 milliseconds with 0 bytes received, result=

Does someone know a solution to this?

**Comment by scamille** [comment]
Are you using a prebuilt simc docker image, or building it yourself? If it is the latter, it is probably easier to bake it into the docker image buy using the option here: https://github.com/simulationcraft/simc/blob/dragonflight/Dockerfile#L28

The logic for the .simc_apikey file that is loaded at runtime is here: https://github.com/simulationcraft/simc/blob/f3e2ce8a879dcfff6d55e5af2ef402440ba23d16/engine/sim/sim.cpp#L1120
```cpp
  // unix home
  if ( char* home_path = getenv( "HOME" ) )
  {
    std::string home_apikey(home_path);
    home_apikey += "/.simc_apikey";
    key_locations.push_back( home_apikey );
  }
```
Picks up the .simc_apikey file relative to the HOME env variable.

The other solution is
```cpp
  std::vector<std::string> key_locations;
  key_locations.emplace_back("./apikey.txt" );
```
Just put a .apikey.txt in the working directory of the simc executable. While this was originally intended for windows clients I think, it might be the simpler solution for Docker.
